### PR TITLE
Reduce hardcoded locale configuration

### PR DESCRIPTION
**Pull request by albertchae**
Issue: https://github.com/rubyforgood/terrastories/issues/13

Adding a new locale just means making a new directory under
`config/locales` with the appropriate yaml files. The one thing that
must still be changed in the code is `default_locale`.

### CODE REVIEW

File: rails/app/views/welcome/index.html.erb [hunk 1]
```diff
@@ -23,8 +23,12 @@
     </div>
 
     <div class="language-picker">
-      <%= t("language") %>: <%= link_to "mat (#{t("default")})", root_path(locale: "mat") %> |
-      <%= link_to "en", root_path(locale: "en") %>
+      <%= t("language") %>:
+      <%= link_to "#{I18n.default_locale} (#{t("default")})", root_path(locale: I18n.default_locale) %> |
+      <% non_default_locales = (I18n.available_locales - [I18n.default_locale]).map do |locale| %>
```
Comment (baweaver) [inline rails/app/views/welcome/index.html.erb]: Can we move `non_default_locales` into a helper?

File: rails/config/routes.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@
     root to: "users#index"
   end
 
-  scope "(:locale)", locale: /en|mat/ do
+  scope "(:locale)", locale: /#{I18n.available_locales.join('|')}/ do
```
Comment (baweaver) [inline rails/config/routes.rb]: `Regexp.union` may be a bit more explicit here:

```ruby
scope '(:locale)', locale: Regexp.union(I18n.available_locales) do
```

File: rails/app/views/welcome/index.html.erb (hunk 1 above)
Comment (albertchae) [reply rails/app/views/welcome/index.html.erb]: 👍

File: rails/config/routes.rb (hunk 2 above)
Comment (albertchae) [reply rails/config/routes.rb]: Cool, didn't know about `Regexp.union`

**Pull request merged**